### PR TITLE
Fixes 2550 by changing the order if the leader got elected.

### PR DESCRIPTION
**Pull request by aquamatthias**
The second problem is the failing caching mechanism, when the prefix is empty (as this is the case for frameworkId)
Added a migration from frameworkId->framework:id

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/MarathonSchedulerDriver.scala [hunk 1]
```diff
@@ -17,6 +17,9 @@ object MarathonSchedulerDriver {
                 httpConfig: HttpConf,
                 newScheduler: MarathonScheduler,
                 frameworkId: Option[FrameworkID]): SchedulerDriver = {
+
+    log.info(s"Create new Scheduler Driver with frameworkId: $frameworkId")
```
Comment (kolloch) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerDriver.scala]: Thanks! Nice addition :+1:

File: src/main/scala/mesosphere/marathon/state/EntityStoreCache.scala [hunk 2]
```diff
@@ -83,6 +83,7 @@ class EntityStoreCache[T <: MarathonState[_, T]](store: EntityStore[T])
 
     def handleEntries(names: Seq[String]): Future[Unit] = {
       val (unversionedNames, versionedNames) = names.partition(noVersionKey)
+      log.debug(s"$store Preload and cache entries: $unversionedNames and versioned entries $versionedNames")
```
Comment (kolloch) [inline src/main/scala/mesosphere/marathon/state/EntityStoreCache.scala]: These can result in a lot of "names". So please wrap the debug statement in an `if (log.isDebugEnabled)`.

(We should really switch to suitable Scala logging macros)

Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/state/EntityStoreCache.scala]: Done - and I agree. Any proposals?

**Comment by kolloch** [comment]
LGTM!

**Pull request merged**